### PR TITLE
Config as Code Reference and more doco

### PR DESCRIPTION
**Pull request by BobJWalker**
This PR further breaks out the config as code documentation page and adds in a lot more detail to help customers get going.

### CODE REVIEW

File: docs/projects/version-control/configuring-version-control-on-a-project.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+:::hint
+You can have multiple deployment processes in the same repository if they all use a different sub-directory.
+:::
+
+Once you press the **Configure** button, a modal window will appear to confirm this change and give you the option to provide a summary and description for the first commit or cancel the conversion.
+
+![configuring version control](configure-version-control.png)
+
+Your project is now configured with Version Control. You can see this change reflected on the Process page, where you can change branches. You can also confirm this in your Git repository. The `.octopus` directory will not be created, and it should contain your _deployment_process.ocl_, _deployment_settings.ocl_, and _schema_version.ocl_ files.
```
Comment (harrisonmeister) [inline docs/projects/version-control/configuring-version-control-on-a-project.md:53]:
```suggestion
Your project is now configured with Version Control. You can see this change reflected on the Process page, where you can change branches. You can also confirm this in your Git repository. The `.octopus` directory will now be created, and it should contain your _deployment_process.ocl_, _deployment_settings.ocl_, and _schema_version.ocl_ files. 
```

File: docs/projects/version-control/configuring-version-control-on-a-project.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+The Configuration as Code feature is per-project. Currently, only the deployment process and settings are saved to version control. Eventually, it will include variables and runbooks. 
+
+Even after variables and runbooks are added, a number of project-level and instance-level settings will not be stored in version control.
+
+Learn more about [what is stored in version control](/docs/projects/version-control/config-as-code-reference.md).
+
+## Using a project with version control enabled
+
+In general, modifying a project via the Octopus UI with version control enabled is the same as modifying a project configured to save changes to SQL Server. There are some minor differences.
```
Comment (harrisonmeister) [inline docs/projects/version-control/configuring-version-control-on-a-project.md:65]:
```suggestion
In general, modifying a project via the Octopus UI with version control enabled is the same as modifying a project configured to save changes to SQL Server. However, there are some minor differences.
```

File: docs/projects/version-control/config-as-code-reference.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+position: 20 
+---
+
+The configuration as code feature enables you to save some project-level settings as files in a git repository instead of SQL Server. The files are written in OCL (Octopus Configuration Language) format. Storing resources as files enables you to leverage version control features such as branching, pull requests, and reverting. In addition, how you deploy your code and your source code can now be saved in the same git repository. This page is a reference document of how the config-as-code feature works.
+
+## Octopus Project Level Only
+
+The config-as-code feature will store Octopus Project resources in git instead of SQL Server.
+
+### Resources NOT version controlled by config-as-code
```
Comment (harrisonmeister) [inline docs/projects/version-control/config-as-code-reference.md:13]: I think this section on resources that are not version-controlled might be better moved to after the ones that are?

File: docs/projects/version-control/config-as-code-reference.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+connectivity_policy {
+}
+
+versioning_strategy {
+}
+```
+
+However, configuring the settings via Octopus will populate the fields with their properties and values.
+
+```
```
Comment (harrisonmeister) [inline docs/projects/version-control/config-as-code-reference.md:188]:
```suggestion
```text
```

File: docs/projects/version-control/config-as-code-reference.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+- The Octopus Terraform Provider and OCL are not a 1:1 match. You cannot copy resources between the two and expect everything to work. We want to narrow the gap as much as possible, but as of right now, a gap exists.
+- Shared resources (environments, external feeds, step templates, etc.) are referenced by name, not ID. Be very careful when changing the names of any shared resources.  
+- Shared resources must exist before loading an OCL file into Octopus Deploy. What that means is if you copy the OCL files from one git repo to another, and point a new project at those files, then any shared resource must exist before creating that project. That only applies when projects are in different spaces or on different instances. If the resources do not exist, an error message will appear.
+- Pointing multiple projects to the same folder in the same git repo is unsupported. Please see [unsupported config as code scenarios](/docs/projects/version-control/unsupported-config-as-code-scenarios.md) for more information.
```
Comment (harrisonmeister) [inline docs/projects/version-control/config-as-code-reference.md:228]:
```suggestion
- Pointing multiple projects to the same folder in the same git repo is unsupported. Please see our [unsupported config as code scenarios](/docs/projects/version-control/unsupported-config-as-code-scenarios.md) for more information.
```

File: docs/projects/version-control/editing-a-project-with-version-control-enabled.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+---
+title: Editing a project with version control enabled
+description: What to expect when using the Configuration as Code feature in Octopus Deploy
+position: 30 
+---
+
+Once an Octopus Project is configured to be version-controlled, your experience making changes to a project will change. With the configuration as code feature, you can either edit the resources via the Octopus Deploy UI or your favorite file edit tool. This page will walk through what to expect.
```
Comment (harrisonmeister) [inline docs/projects/version-control/editing-a-project-with-version-control-enabled.md:7]:
```suggestion
Once an Octopus Project is configured to be version-controlled, your experience making changes to a project will change. With the configuration as code feature, you can either edit the resources via the Octopus Deploy UI or your favorite file editing tool. This page will walk through what to expect.
```

File: docs/projects/version-control/editing-a-project-with-version-control-enabled.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+![Branch-switcher user-interface](branch-switcher-ui.png "width=500")
+
+Each branch can have a different deployment process. For example, if you decide to move from running web applications on VMs to PaaS, you could create a branch to hold all your code changes and your deployment process changes. You can make the necessary updates to your deployment process to deploy to a PaaS target. The deployment process currently being used to deploy to **Production** can be left alone until you are ready to merge in both the code changes and the deployment process changes.
+
+### Commits
+
+Before enabling version control on the project, clicking on save updated a record in the SQL Server database. That record would be overwritten with each change. Once version control is enabled, that is no longer the case. You'll be generating commits each time you "save."  As such, the save experience has been updated. The **Save** button has been replaced with **Commit**, and clicking on that will allow you to enter a commit message before saving. Next to the **Commit** button is a quick save, useful when you made a minor change.
```
Comment (harrisonmeister) [inline docs/projects/version-control/editing-a-project-with-version-control-enabled.md:23]:
```suggestion
Before enabling version control on the project, clicking on save updated a record in the SQL Server database. That record would be overwritten with each change. Once version control is enabled, that is no longer the case. You'll be generating commits each time you "save."  As such, the save experience has been updated. The **Save** button has been replaced with **Commit**, and clicking on that will allow you to enter a commit message before saving. Next to the **Commit** button is a quick save, useful when you make a minor change.
```

File: docs/projects/version-control/editing-a-project-with-version-control-enabled.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+![committing a change to version control](commit-process.png)
+
+### Viewing and Editing OCL
+
+Enabling version control also adds the option to **View/Edit OCL** to the `...` or overflow menu. Clicking on this option will open up a modal window allowing you to edit the OCL (Octopus Configuration Language) file directly.
```
Comment (harrisonmeister) [inline docs/projects/version-control/editing-a-project-with-version-control-enabled.md:29]:
```suggestion
Enabling version control also adds the option to **View/Edit OCL** to the overflow menu (`...`). Clicking on this option will open up a modal window allowing you to edit the OCL (Octopus Configuration Language) file directly.  
```

File: docs/projects/version-control/editing-a-project-with-version-control-enabled.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+:::
+
+Once you are finished editing the OCL file, you can commit changes those changes to the git repo. The UI will automatically refresh.
+
+## Editing via a text editor
+
+Changes can also be made using your favorite text editor or IDE and committed and pushed just as you would any other code change. 
+
+:::hint
+We have a [Visual Studio Code Plug-in](https://marketplace.visualstudio.com/items?itemName=octopusdeploy.vscode-octopusdeploy) that will add syntax highlighting, OCL snippets, and an integrated tree view for navigating nodes in an HCL file.
```
Comment (harrisonmeister) [inline docs/projects/version-control/editing-a-project-with-version-control-enabled.md:44]: Wasn't sure if this should be OCL (vs HCL) ?

```suggestion
We have a [Visual Studio Code Plug-in](https://marketplace.visualstudio.com/items?itemName=octopusdeploy.vscode-octopusdeploy) that will add syntax highlighting, OCL snippets, and an integrated tree view for navigating nodes in an OCL file.
```

**Review by harrisonmeister — COMMENTED** [review]
A few minor suggestions, looks really good though :+1:

File: docs/projects/version-control/editing-a-project-with-version-control-enabled.md (hunk 9 above)
Comment (BobJWalker) [reply docs/projects/version-control/editing-a-project-with-version-control-enabled.md:44]: It's OCL.

File: docs/projects/version-control/index.md [hunk 10]
```diff
@@ -12,83 +12,67 @@ The Early Access Preview of the config-as-code feature was added in **Octopus 20
 
 ## Introduction 
 
-Welcome to the config-as-code Early Access Preview!  Support for version-controlling Octopus projects has been highly requested for a long time now, and we're excited to release the first cut. The goal of the EAP is to gather feedback, and help us evaluate which problems we have solved, and which are the most valuable to solve next.
```
[the remark this answers is not in the document]
Comment (BobJWalker) [reply docs/projects/version-control/index.md:15]: Agreed, updated

**Pull request merged**